### PR TITLE
Check account disk quotas before writing using to_carto

### PR DESCRIPTION
**Pull request by antoniocarlon**
Check account disk quotas before writing using `to_carto`

**Note:** The PR is blocked until we decide what to do with in discussion here https://app.clubhouse.io/cartoteam/story/93719/check-account-disk-quotas-before-writing

### CODE REVIEW

File: cartoframes/io/carto.py [hunk 1]
```diff
@@ -89,6 +91,9 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
             uses the name of the index from the dataframe.
         cartodbfy (bool, optional): convert the table to CARTO format. Default True. More info
             `here <https://carto.com/developers/sql-api/guides/creating-tables/#create-tables>`.
+        ignore_quota_warning (bool, optional): ignore the warning of the possible quota exceeded
+            and force the upload.
```
Comment (Jesus89) [inline cartoframes/io/carto.py:95]: I would add here `Default False.` to be consistent with the other descriptions

File: cartoframes/io/carto.py [hunk 2]
```diff
@@ -114,6 +119,17 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
 
     context_manager = ContextManager(credentials)
 
+    if not ignore_quota_warning:
```
Comment (Jesus89) [inline cartoframes/io/carto.py:122]: We could extract here `me_data = context_manager.credentials.me_data` to have a bit more readable code.

File: cartoframes/io/carto.py [hunk 3]
```diff
@@ -114,6 +119,17 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
...
+        if context_manager.credentials.me_data is not None and context_manager.credentials.me_data.get('user_data'):
+            n = min(SAMPLE_ROWS_NUMBER, len(dataframe))
+            dataframe_size = len(dataframe.sample(n=n).to_csv(header=False)) * len(dataframe) / n / CSV_TO_CARTO_RATIO
```
Comment (Jesus89) [inline cartoframes/io/carto.py:125]: I would rename this `dataframe_size` to `estimated_byte_size`

File: cartoframes/io/carto.py [hunk 4]
```diff
@@ -89,6 +91,9 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
             uses the name of the index from the dataframe.
         cartodbfy (bool, optional): convert the table to CARTO format. Default True. More info
             `here <https://carto.com/developers/sql-api/guides/creating-tables/#create-tables>`.
+        ignore_quota_warning (bool, optional): ignore the warning of the possible quota exceeded
```
Comment (Jesus89) [inline cartoframes/io/carto.py:94]: I think we can use `skip` instead of `ignore` in the param name. What do you think?

**Review by Jesus89 — CHANGES_REQUESTED** [review]
Good job. Some minor comments added. I miss also two units tests to cover the new feature (working and exception raised)

File: cartoframes/io/carto.py [hunk 5]
```diff
@@ -114,6 +119,19 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
 
     context_manager = ContextManager(credentials)
 
+    if not skip_quota_warning:
+        me_data = context_manager.credentials.me_data
+        if me_data is not None and me_data.get('user_data'):
+            n = min(SAMPLE_ROWS_NUMBER, len(dataframe))
+            estimated_byte_size = len(dataframe.sample(n=n).to_csv(header=False)) * len(dataframe) \
+                / n / CSV_TO_CARTO_RATIO
+            remaining_byte_quota = context_manager.credentials.me_data.get('user_data').get('remaining_byte_quota')
```
Comment (Jesus89) [inline cartoframes/io/carto.py:128]: One more `me_data` here

**Comment by antoniocarlon** [comment]
@Jesus89 second CR round?

**Review by Jesus89 — APPROVED** [review]
LGTM!

For the acceptance, could you try an auto-generated CSV file to test the real limits?

**Pull request merged**